json_annotation/lib/src/allowed_keys_helpers.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Helper function used in generated code when
+/// `JsonAnnotation.allowUnrecognizedKeys` is `false`.
+///
+/// Should not be used directly.
+void $checkAllowedKeys(Map map, Iterable<String> allowedKeys) {
+  var invalidKeys = map.keys.where((k) => !allowedKeys.contains(k));
```

Comment:
Might as well add `.toList` here. If there are no keys, then you create an empty list – cheap.

Then we avoid hitting an iterator twice – which is something we should model...